aws/resource_aws_eip.go

```diff
@@ -260,12 +260,12 @@ func resourceAwsEipRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	d.Set("association_id", address.AssociationId)
-	if address.InstanceId != nil {
+	if address.InstanceId != nil { // nosem: helper-schema-ResourceData-Set-extraneous-nil-check
```

Comment:
Reverse these tests too?

Reply:
In fact setting the resource data field to `""` is what happens if the passed in value is `nil`, so the `else` branch is not needed.